charts/kubernetes-agent/templates/tentacle-deployment.yaml

```diff
@@ -51,22 +51,26 @@ spec:
               value: {{ .Values.agent.serverUrl | quote }}
             - name: "Space"
               value: {{ .Values.agent.space | quote }}
+            - name: "RegisterAs"
+              value: {{ .Values.agent.registerAs | quote }}
             - name: "TargetEnvironment"
-              value: {{ join "," .Values.agent.targetEnvironments | quote }}
+              value: {{ join "," .Values.agent.deploymentTarget.environments | quote }}
             - name: "TargetRole"
-              value: {{ join "," .Values.agent.targetRoles | quote }}
+              value: {{ join "," .Values.agent.deploymentTarget.roles | quote }}
             - name: "TargetTenant"
-              value: {{ join "," .Values.agent.targetTenants | quote }}
+              value: {{ join "," .Values.agent.deploymentTarget.tenants | quote }}
             - name: "TargetTenantTag"
-              value: {{ join "," .Values.agent.targetTenantTags | quote }}
+              value: {{ join "," .Values.agent.deploymentTarget.tenantTags | quote }}
             - name: "TargetTenantedDeploymentParticipation"
-              value: {{ .Values.agent.targetTenantedDeploymentParticipation | quote }}
+              value: {{ .Values.agent.deploymentTarget.tenantedDeploymentParticipation | quote }}
+            - name: "WorkerPoolIds"
```

Comment:
I figured that there is no harm in passing through these environment variables all the time regardless of the selected registration option. Since the `configure-and-run.sh` script on Tentacle will need to respond to an explicit selector field anyway, and simply passing through here removes the need for another conditional block in the helm template.